### PR TITLE
Harden addRecipientEmail function

### PR DESCRIPTION
**Pull request by peter279k**
### Short description of what this PR does:
- Refactor and enhance the conditions.
- It's also related to issue #614.
- I also use `PHP_CodeSniffer` to fix the coding style with `PSR-2`.

### CODE REVIEW

**Comment by SendGridDX** [comment]
[CLA assistant check](https://cla.sendgrid.com/sendgrid/sendgrid-php?pullRequest=715) <br/>Thank you for your submission, we really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla.sendgrid.com/sendgrid/sendgrid-php?pullRequest=715) before we can accept your contribution.<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla.sendgrid.com/check/sendgrid/sendgrid-php?pullRequest=715) it.</sub>

File: lib/mail/Mail.php [hunk 1]
```diff
@@ -214,28 +214,23 @@ private function addRecipientEmail(
...
-                $this->personalization[0]->$personalizationFunctionCall($email);
-                if ($subs = $email->getSubstitutions()) {
-                    foreach ($subs as $key => $value) {
-                        $this->personalization[0]->addSubstitution($key, $value);
-                    }
-                }
-                return;
-            } else if ($this->personalization[$personalizationIndex] != null) {
+
+            $personalizationIndex = ($personalizationIndex === null ? 0 : $personalizationIndex);
```
Comment (peter279k) [inline lib/mail/Mail.php]: Using `?:` syntax to merge the two duplicate conditions.

File: lib/mail/Mail.php [hunk 2]
```diff
@@ -214,28 +214,23 @@ private function addRecipientEmail(
             if (isset($personalizationIndex)
                 && !isset($this->personalization[$personalizationIndex])
             ) {
-                // TODO: We should only do this if there exists an index
-                // previous. For example, if given an index 3 and there is
-                // no index 2, we should throw an error.
+                if ($personalizationIndex !== 0 &&
```
Comment (peter279k) [inline lib/mail/Mail.php]: Add this `if-condition` block for `TODO` feature.

**Comment by thinkingserious** [comment]
Hello @peter279k,<br /><br/>Thanks again for the PR!<br />                                              <br />It's <a href="https://sendgrid.com/blog/hacktoberfest-2018-has-arrived/">HACKTOBERFEST</a>! We want to show our appreciation by sending                                              you some special Hacktoberfest swag. If you have not already, could you please fill out                                              <a href="https://dx.sendgrid.com/hacktoberfest/join">this form</a> so we can send it to you? Thanks!                                             <br /><br />Team SendGrid DX

**Comment by childish-sambino** [comment]
Replaced by https://github.com/sendgrid/sendgrid-php/pull/952

**Pull request closed without merging**